### PR TITLE
feat: pi05 batch all camera images into single vision encoder call

### PR DESCRIPTION
**Pull request by eugene123tw**
## Description

- **Eval / tracing**: batches all camera images into one encoder call (`[N*B, C, H, W]`), avoiding redundant sequential vision encoder invocations.
- **Training**: iterates per-camera with gradient checkpointing support.

## Type of Change

- [x] ✨ `feat` - New feature
- [ ] ♻️ `refactor` - Code refactoring
- [ ] 🐞 `fix` - Bug fix
- [ ] 📚 `docs` - Documentation
- [ ] 🧪 `test` - Tests
- [ ] 🔧 `chore` - Maintenance

## Changes Made

| File | Change |
|------|--------|
| `library/src/physicalai/policies/pi05/model.py` | Add images batching; added `.bool()` cast in `_prepare_attention_masks_4d` for JIT tracing compatibility |
| `library/tests/unit/policies/test_pi05.py` | Added `TestEmbedPrefix` class with 8 unit tests covering shapes, encoder call count, mask behavior, and train/eval parity |

## Examples

```python
def embed_prefix(self, ...):
    use_batched = not self.training or torch.jit.is_tracing() or torch.onnx.is_in_onnx_export()
    if use_batched:
        # single batched encoder call
    for cam_idx in range(num_cameras):
        if use_batched:
            img_emb = all_img_embs[cam_idx]
        else:
            img_emb = self._apply_checkpoint(image_embed_func, images[cam_idx])
    ...
```

### CODE REVIEW

File: library/src/physicalai/policies/pi05/model.py [hunk 1]
```diff
@@ -821,45 +823,65 @@ def embed_prefix(
...
+            tokens: ``(batch, seq_len)`` tokenized prompt.
+            masks: ``(batch, seq_len)`` prompt attention mask.
+
         Returns:
             Tuple of (embeddings, padding masks, attention masks).
         """
-        embs = []
-        pad_masks = []
-        att_masks = []
+        use_batched = not self.training or torch.jit.is_tracing() or torch.onnx.is_in_onnx_export()
```
Comment (sovrasov) [inline library/src/physicalai/policies/pi05/model.py]: The latter checks might be a bit extra, because we don't export in training mode: that would result in exporting loss computation

**Review by sovrasov — APPROVED** [review]
